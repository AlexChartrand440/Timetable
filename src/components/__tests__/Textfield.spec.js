// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MKTextField } from 'react-native-material-kit';
import Textfield from '../Textfield';

const props = {
  input: {
    onChange: jest.fn(),
    value: 'some value',
  },
  value: 'another value',
};

describe('Textfield component', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Textfield {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error messages correctly', () => {
    const otherProps = {
      ...props,
      meta: {
        error: 'an error message',
        submitFailed: true,
        touched: true,
      },
    };
    const tree = renderer.create(<Textfield {...otherProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses props.input.value as value if defined', () => {
    const wrapper = shallow(<Textfield {...props} />);
    const textfield = wrapper.find(MKTextField);
    expect(textfield.prop('value')).toBe(props.input.value);
  });

  it('uses props.value as value if props.input.value is not defined', () => {
    const otherProps = {
      input: {
        onChange: () => { },
      },
      value: 'another value',
    };
    const wrapper = shallow(<Textfield {...otherProps} />);
    const textfield = wrapper.find(MKTextField);
    expect(textfield.prop('value')).toBe(otherProps.value);
  });

  it('calls input.onChange with new input when input changes', () => {
    const wrapper = shallow(<Textfield {...props} />);
    const textfield = wrapper.find(MKTextField);
    textfield.simulate('changeText', 'new text');
    const { onChange } = props.input;
    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange.mock.calls[0][0]).toBe('new text');
  });
});
