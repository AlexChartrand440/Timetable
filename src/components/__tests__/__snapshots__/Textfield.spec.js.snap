// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={undefined}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      secureTextEntry={undefined}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "flex": 1,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="some value"
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 1,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#247BA0",
            "height": 1,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Textfield component renders error messages correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={undefined}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      secureTextEntry={undefined}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "flex": 1,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="some value"
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
            "height": 1,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#247BA0",
            "height": 1,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    an error message
  </Text>
</View>
`;
